frontend/packages/knative-plugin/src/components/add/EventSourcePage.tsx

```diff
@@ -10,20 +10,44 @@ import NamespacedPage, {
 import { QUERY_PROPERTIES } from '@console/dev-console/src/const';
 import ConnectedEventSource from './EventSource';
 import EventSourceAlert from './EventSourceAlert';
-import { useEventSourceList } from '../../utils/create-eventsources-utils';
+import { useEventSourceList, useGetKamelet } from '../../utils/create-eventsources-utils';
 import { isDynamicEventSourceKind } from '../../utils/fetch-dynamic-eventsources-utils';
+import { CamelKameletBindingModel } from '../../models';
 
 type EventSourcePageProps = RouteComponentProps<{ ns?: string }>;
 
 const EventSourcePage: React.FC<EventSourcePageProps> = ({ match, location }) => {
   const { t } = useTranslation();
   const namespace = match.params.ns;
-  const eventSourceStatus = useEventSourceList(namespace);
+  const eventSourceList = useEventSourceList(namespace);
   const searchParams = new URLSearchParams(location.search);
   const showCatalog = location.pathname.includes('/extensible-catalog/');
   const sourceKindProp = showCatalog && searchParams.get('sourceKind');
-  const isSourceKindPresent = sourceKindProp && isDynamicEventSourceKind(sourceKindProp);
-
+  const kameletName = showCatalog && sourceKindProp && searchParams.get('kameletName');
+  const [kamelet, kameletLoaded] = useGetKamelet(kameletName, namespace);
+  const eventSourceStatus = {
+    ...eventSourceList,
+    ...(sourceKindProp === CamelKameletBindingModel.kind
+      ? { loaded: eventSourceList.loaded || kameletLoaded }
+      : { loaded: eventSourceList.loaded }),
```

Comment:
does this make sense

```suggestion
    ...(kameletName && kameletLoaded
      ? { loaded: kameletLoaded }
      : { loaded: eventSourceList.loaded }),
```